chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/Configurations.scala

```diff
@@ -0,0 +1,149 @@
+          value: Expr[ToField]
+      ) extends ValueSource[From, ToField]
+
+      final case class ConstPartial[From, ToField](
+          value: Expr[partial.Result[ToField]]
+      ) extends ValueSource[From, ToField]
+
+      final case class Computed[From, ToField](
+          compute: Expr[From] => Expr[ToField]
+      ) extends ValueSource[From, ToField]
+
+      final case class ComputedPartial[From, ToField](
+          compute: Expr[From] => Expr[partial.Result[ToField]]
+      ) extends ValueSource[From, ToField]
+
+      final case class Renamed[From, FromField, ToField](
+          fromFieldName: String,
+          fromFieldType: Type[FromField],
+          originalValue: Expr[From] => Expr[FromField]
+      ) extends ValueSource[From, ToField]
+    }
+
+    sealed abstract class RuntimeConfiguration(val needValueLevelAccess: Boolean) extends Product with Serializable
+    object RuntimeConfiguration {
+
+      final case class Const(runtimeDataIdx: Int) extends RuntimeConfiguration(true)
+      final case class ConstPartial(runtimeDataIdx: Int) extends RuntimeConfiguration(true)
+      final case class Computed(runtimeDataIdx: Int) extends RuntimeConfiguration(true)
+      final case class ComputedPartial(runtimeDataIdx: Int) extends RuntimeConfiguration(true)
+      final case class RenamedFrom(sourceName: String) extends RuntimeConfiguration(false)
+    }
+  }
+
+  sealed protected trait CoproductOverride[From, FromSubtype <: From, To] extends Product with Serializable {
+    val fromSubtype: Type[FromSubtype]
+  }
+  protected object CoproductOverride {
+
+    final case class CoproductTotalInstance[From, FromSubtype <: From, To](
+        fromSubtype: Type[FromSubtype],
+        convert: Expr[FromSubtype] => Expr[To]
+    ) extends CoproductOverride[From, FromSubtype, To]
+
+    final case class CoproductPartialInstance[From, FromSubtype <: From, To](
+        fromSubtype: Type[FromSubtype],
+        convert: Expr[FromSubtype] => Expr[partial.Result[To]]
+    ) extends CoproductOverride[From, FromSubtype, To]
+  }
+
+  final protected case class TransformerConfig[From, To](
+      flags: TransformerFlags = TransformerFlags(),
+      fieldOverrides: Map[String, FieldOverride[From, ?]] = Map.empty,
+      coproductOverride: Vector[CoproductOverride[From, ? <: From, To]] = Vector.empty,
+      legacy: TransformerConfig.LegacyData = TransformerConfig.LegacyData() // TODO: temporary
+  )
+  object TransformerConfig {
+
+    // TODO: for creating TransformerConfig for old macros in Scala 2 until everything is migrated
+    final case class LegacyData(
+        fieldOverrideLegacy: Map[String, FieldOverride.RuntimeConfiguration] = Map.empty,
+        coproductInstanceOverridesLegacy: Map[(ComputedType, ComputedType), Int] = Map.empty,
+        coproductInstancesPartialOverridesLegacy: Map[(ComputedType, ComputedType), Int] = Map.empty,
+        transformerDefinitionPrefix: Expr[dsls.TransformerDefinitionCommons.RuntimeDataStore] =
```

Comment:
In current master we have:

```
      transformerDefinitionPrefix: Tree = EmptyTree,
      definitionScope: Option[(Type, Type)] = None
```

We're probably going to need those in order to preserve current behavior. And not talking only about legacy config, but also in the new one.

Reply:
They are already stored in `Context[From, To]` and function converting to legacy config takes a whole context.

Reply:
they were not immediately, but they were added in the process... 👍